### ISSUE TITLE
When editing program, my modifications are sometimes applied to different tabs (specifically the rightmost tab) instead. bug?

### ISSUE THREAD

**Issue by Timpin2001** [Pulover/PuloversMacroCreator#183]
Good evening,
 I have been experiencing an issue with editing my code when I want to make a new macro or change an existing one. If I ever have multiple tabs open within a single macro file, then every time I go to change or add something in one of my tabs, it often will instead apply the change to the same _line_ of code, but the line of code in a _different tab_ instead. I will use some screenshots as an example:
![Screenshot 2021-05-18 073805](https://user-images.githubusercontent.com/83847946/118647541-7250f480-b7af-11eb-9f6c-0bf38b7fc99d.png)
![Screenshot 2021-05-18 073809](https://user-images.githubusercontent.com/83847946/118647559-78df6c00-b7af-11eb-904f-17fbd9742c4b.png)

In this, let's say I go to line 3 of Skeletons and try to change it from setting %DawdleAmt% to 3 to setting %DawdleAmt% to 4. When I change it and hit Apply, it may at first seem as though nothing happened. However, if I flip over to the OrcGrinding tab, I may notice that, all of a sudden, %DawdleAmt% in line 3 is being set to 4 instead of 3, while Skeletons is still setting it to 3. Or, another example; let's say that after line 9 in Skeletons, I wanted to have it then perform a GoSub function to a block of code in the Distillery tab labelled RestBoth. I would select line 10, then go to the Add Goto/Label function on the program, and tell it to make a GoSub function travelling to RestBoth. I would then hit the Okay button, and nothing would seem to happen. However, I flip to the OrcGrinding tab, and what do you know? It now has a GoSub function to RestBoth underneath the LoopEnd line. This issue has been occurring ever since I started using multiple tabs, and I have been able to find a pattern to it: whenever I make a change, it has a chance to make that change to the same line of code in the rightmost tab instead of the intended one. This happens maybe 1 in every 5 times I try to make a change, and I have tried to see if maybe it was a matter of me having last selected a line of code in the different tab, but it doesn't seem to ever change anything other than the specific line that I am trying to change in the other tab, so that cannot be it. I have not been able to find a pattern to it other than this, and I do not believe it is intended. It has been bothering me for a few days now, and I do not know how to change it. The only successful stopgap I have found so far is to take whatever tab I'm working in and move it to become the rightmost tab, which seems to alleviate the issue until I decide to make a change to a different tab. I am unsure why this happens, and I am desperately hoping this is some setting that I can change so that I don't have to deal with this issue.

 Thanks,
  ~Timpin17 (She/Her)

**Comment by Timpin2001** [comment]
Minor amendment: my mother walked by after I had posted this, and I wanted to show her what I was talking about by actually making it happen in front of her. I had decided to do it by adding a few different GoTo functions to the Skeletons tab, and the first four attempts worked properly, but the fifth attempt resulted in the bug mentioned above. However, it did not make the change to the same line of code, it made the change to a different one. However, I have had it affect seemingly random lines of code before, so it seems like it's either assigning them to _random_ lines of code each time, or it will sometimes be a change to the same line and sometimes will be a change to a seemingly random line. I do not know what is making it happen, but I just wanted to share that it does not seem to actually _always_ affect the same line of code. However, it does seem to do so _sometimes_.

**Comment by TirOFlanc** [comment]
Hi,

It's the same problem as this one:
https://github.com/Pulover/PuloversMacroCreator/issues/155#issue-782085350

**Comment by Timpin2001** [comment]
You are correct! I did not notice that you had already made a discussion on this issue, thank you. I did not see any solutions get presented, do you know of any solutions you had found for it?

**Comment by TirOFlanc** [comment]
Nop !

Sorry :(

**Comment by higrm** [comment]
I've seen this myself, code appearing in the rightmost tab instead of where intended, but as mentioned above, it has some degree of randomness to it, so I cannot give steps to reproduce.
